reactos/drivers/usb/usbport/device.c

```diff
@@ -0,0 +1,1873 @@
+#include "usbport.h"
+
+#define NDEBUG
+#include <debug.h>
+
+NTSTATUS
+NTAPI
+USBPORT_SendSetupPacket(IN PUSBPORT_DEVICE_HANDLE DeviceHandle,
+                        IN PDEVICE_OBJECT FdoDevice,
+                        IN PUSB_DEFAULT_PIPE_SETUP_PACKET SetupPacket,
+                        IN PVOID Buffer,
+                        IN ULONG Length,
+                        IN OUT PULONG TransferedLen,
+                        IN OUT PUSBD_STATUS pUSBDStatus)
+{
+    PURB Urb;
+    PMDL Mdl;
+    USBD_STATUS USBDStatus;
+    KEVENT Event;
+    NTSTATUS Status;
+
+    DPRINT("USBPORT_SendSetupPacket: DeviceHandle - %p, FdoDevice - %p, SetupPacket - %p, Buffer - %p, Length - %x, TransferedLen - %x, pUSBDStatus - %x\n",
+           DeviceHandle,
+           FdoDevice,
+           SetupPacket,
+           Buffer,
+           Length,
+           TransferedLen,
+           pUSBDStatus);
+
+    KeInitializeEvent(&Event, NotificationEvent, FALSE);
+
+    Urb = (PURB)ExAllocatePoolWithTag(NonPagedPool,
```

Comment:
Yes, for a computer this is not of interest. Only for human eyes. Deleted all same casts.